operator/src/main/java/oracle/kubernetes/operator/logging/LoggingFilter.java

```diff
@@ -0,0 +1,35 @@
+// Copyright 2019, Oracle Corporation and/or its affiliates.  All rights reserved.
+// Licensed under the Universal Permissive License v 1.0 as shown at
+// http://oss.oracle.com/licenses/upl.
+
+package oracle.kubernetes.operator.logging;
+
+/** A filter to control whether a log message should be logged */
+public interface LoggingFilter {
+
+  // Constant for key for storing LoggingFilter object in Packet map
+  String LOGGING_FILTER_PACKET_KEY = "loggingFilter";
+
+  /**
+   * Checks if the message should be logged
+   *
+   * @param msg the message to be loggged
+   */
+  boolean canLog(String msg);
+
+  /**
+   * Checks if the message should be logged according to an optional LoggingFilter
+   *
+   * @param loggingFilter LoggingFilter that decides whether the log message should be logged, can
+   *     be null
+   * @param msg The log message to be loggd
+   * @return the canLog() return value from the provided loggingFilter, message, and parameters, or
+   *     true if loggingFilter is null
+   */
+  static boolean canLog(LoggingFilter loggingFilter, String msg) {
+    if (loggingFilter == null) {
```

Comment:
You can simplify this as:
```
return loggingFilter == null || loggingFilter.canLog();
```
or, if you don't like those kinds of conditionals:
```
return Optional.ofNullable(loggingFilter).map(loggingFilter::canLog).orElse(true);
```
although I prefer the first as simpler. In either case, it should be easier for most programmers to understand with less reading.

Reply:
I prefer the first too as it is easier to read. will make the change. Thanks